department-of-veterans-affairs/caseflow
# Report RequestIssue stats to DataDog

Resolves #13769

### Description
Provides initial capability to run a new job that reports RequestIssues stats to DataDog. This PR will be an example for future metrics, which can be added to this job (if it's related) or implemented as a new job.

A StatsCollectorJob has been implemented to simplify sending metrics to DataDog and to minimize the cost of running jobs as AWS Lambdas. StatsCollectorJob holds sets of Collectors that run daily, weekly, and monthly. 

The StatsCollectorJob is scheduled to run in a Lambda in https://github.com/department-of-veterans-affairs/appeals-lambdas/pull/51.

### Acceptance Criteria
- [x] A single StatsCollectorJob that runs daily, weekly, and monthly `Collector`s.

### Testing Plan (after merge)
1. Ensure no related errors appear in #appeals-job-alerts
1. Check DataDog for reported stats
1. Have the job run daily to make sure it's working. 
1. Once confirmed that things are working, change the `RequestIssuesStatsCollector` to run monthly.
